frame/benchmark/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+		MyValue: u32;
+		MyMap: map hasher(blake2_256) u32 => u32;
+		MyDoubleMap: double_map hasher(blake2_256) u32, hasher(blake2_256) u32 => u32;
+	}
+}
+
+// The pallet's events
+decl_event!(
+	pub enum Event<T> where AccountId = <T as system::Trait>::AccountId {
+		Dummy(u32, AccountId),
+	}
+);
+
+// The pallet's errors
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+	}
+}
+
+// The pallet's dispatchable functions.
+decl_module! {
+	/// The module declaration.
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		/// Do nothing.
+		pub fn do_nothing(_origin, input: u32) {
+			if input > 0 {
+				return Ok(());
+			}
+		}
+
+		/// Read a value from storage value `repeat` number of times.
+		/// Note the first `get()` read here will pull from the underlying
+		/// storage database, however, the `repeat` calls will all pull from the
+		/// storage overlay cache. You must consider this when analyzing the
+		/// results of the benchmark.
+		pub fn read_value(_origin, repeat: u32) {
+			for _ in 0..repeat {
+				MyValue::get();
+			}
+		}
+
+		/// Put a value into a storage value.
+		pub fn put_value(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyValue::put(r);
+			}
+		}
+
+		/// Read a value from storage `repeat` number of times.
+		/// Note the first `exists()` read here will pull from the underlying
+		/// storage database, however, the `repeat` calls will all pull from the
+		/// storage overlay cache. You must consider this when analyzing the
+		/// results of the benchmark.
+		pub fn exists_value(_origin, repeat: u32) {
+			for _ in 0..repeat {
+				MyValue::exists();
+			}
+		}
+
+		/// Remove a value from storage `repeat` number of times.
+		pub fn remove_value(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyMap::remove(r);
+			}
+		}
+
+		/// Read a value from storage map `repeat` number of times.
+		pub fn read_map(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyMap::get(r);
+			}
+		}
+
+		/// Insert a value into a map.
+		pub fn insert_map(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyMap::insert(r, r);
+			}
+		}
+
+		/// Check is a map contains a value `repeat` number of times.
+		pub fn contains_key_map(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyMap::contains_key(r);
+			}
+		}
+
+		/// Read a value from storage `repeat` number of times.
+		pub fn remove_prefix(_origin, repeat: u32) {
+			for r in 0..repeat {
+				MyDoubleMap::remove_prefix(r);
+			}
+		}
+
+		// Add user to the list.
+		pub fn add_member_list(origin) {
+			let who = ensure_signed(origin)?;
+			MyMemberList::<T>::mutate(|x| x.push(who));
+		}
+
+		// Append user to the list.
+		pub fn append_member_list(origin) {
+			let who = ensure_signed(origin)?;
+			MyMemberList::<T>::append(&[who])?;
+		}
+
+		// Append user to the list.
+		pub fn encode_accounts(_origin, accounts: Vec<T::AccountId>) {
+			let _bytes = accounts.encode();
```

Comment:
Are you sure that isn't optimized away?

Reply:
We saw the jumps at powers of two remember?
https://www.shawntabrizi.com/substrate-graph-benchmarks/?p=pallet-benchmark&e=encode_accounts

Is there some trivial syntax I could add to make you feel better here?

Reply:
We could maybe add some check
```
if bytes.empty() {
    panic!("Will never happen, but the compiler can not know this");
}
```

Reply:
turned it into a print statement